src/main/java/com/limechain/storage/block/BlockState.java

```diff
@@ -774,14 +774,18 @@ public void setFinalizedHash(final Hash256 hash, final BigInteger round, final B
      * @throws BlockNodeNotFoundException if the provided setId is less than the highest stored setId.
      */
     public void setHighestRoundAndSetID(final BigInteger round, final BigInteger setId) {
-        final Pair<BigInteger, BigInteger> highestRoundAndSetID = getHighestRoundAndSetID();
-        final BigInteger highestSetID = highestRoundAndSetID.getValue1();
+        try {
+            final Pair<BigInteger, BigInteger> highestRoundAndSetID = getHighestRoundAndSetID();
+            final BigInteger highestSetID = highestRoundAndSetID.getValue1();
 
-        if (setId.compareTo(highestSetID) < 0) {
-            throw new BlockNodeNotFoundException("SetID " + setId + " should be greater or equal to " + highestSetID);
+            if (setId.compareTo(highestSetID) < 0) {
+                throw new BlockStorageGenericException(
+                        "SetID " + setId + " should be greater or equal to " + highestSetID);
+            }
+        } catch (RoundAndSetIdNotFoundException e) {
+            db.save(DBConstants.HIGHEST_ROUND_AND_SET_ID_KEY, helper.bigIntegersToByteArray(round, setId));
```

Comment:
With this change do we save the new setId, when it is greater than the old one?

Reply:
Yes